user-management/build.gradle

```diff
@@ -0,0 +1,298 @@
+            include 'de/tum/in/www1/artemis/repository/**'
+            include 'de/tum/in/www1/artemis/validation/**'
+
+            // de.tum.in.www1.artemis.web.rest
+            include 'de/tum/in/www1/artemis/web/rest/dto/**'
+            include 'de/tum/in/www1/artemis/web/rest/errors/EntityNotFoundException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/BadRequestAlertException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/FieldErrorVM.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/StudentsAlreadyAssignedException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/InternalServerErrorException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/AccessForbiddenException.*'
+            include 'de/tum/in/www1/artemis/web/rest/errors/ErrorConstants.*'
+        }
+        resources {
+            srcDir '../src/main/resources'
+            include 'config/liquibase/**'
+            include 'config/aplication-artemis.yml'
+            include 'config/aplication-gitlab.yml'
+            include 'config/aplication-jenkins.yml'
+        }
+    }
+    externalTest {
+        java {
+            srcDir file("../src/test")
+            include "java/de/tum/in/www1/artemis/util/RequestUtilService.*"
+        }
+    }
+    main {
+        compileClasspath += external.java
+        java {
+            srcDir external.java
+        }
+        resources {
+            srcDir external.resources
+        }
+    }
+    test {
+        compileClasspath += external.java
+        compileClasspath += externalTest.java
+
+        java {
+            srcDir external.java
+            srcDir externalTest.java
+        }
+    }
+}
+
+idea {
+    module {
+        excludeDirs += files("node_modules")
+    }
+}
+
+spotless {
+    // allows to execute the code formatting commands ./gradlew :gateway:spotlessApply and ./gradlew :gateway:spotlessCheck
+    java {
+        target project.fileTree(project.rootDir) {
+            include "gateway/**/*.java"
+            exclude "**/src/main/java/de/tum/in/www1/artemis/usermanagement/config/SecurityConfiguration.java", "**/node_modules/**",
+                "**/out/**", "**/repos/**", "**/build/**", "**/src/main/generated/**", "**/src/main/resources/templates/**",
+                "**/exports/**", "src/main/java/de/tum/in/www1/artemis/**"
+        }
+        importOrderFile "../artemis-spotless.importorder"
+        eclipse().configFile "../artemis-spotless-style.xml"
+
+        removeUnusedImports()
+    }
+}
+
+defaultTasks "bootRun"
+
+springBoot {
+    mainClass = "de.tum.in.www1.artemis.usermanagement.UserManagementApp"
+}
+
+bootRun {
+    jvmArgs = ["--illegal-access=warn"]
+}
+
+modernizer {
+    failOnViolations = true
+    includeTestClasses = true
+}
+
+configurations {
+    providedRuntime
+}
+
+apply from: "../gradle/dependencies.gradle"
+
+dependencies {
+    implementation "com.offbytwo.jenkins:jenkins-client:0.3.8"
```

Comment:
Did you check the dependencies already for duplication in the main build? Just because they seem to be a lot.

Reply:
Yes, that's true. I don't want to extract them to dependencies.gradle because this would mean sharing them with the gateway as well, which is not needed. I think that if we use the versions from the gradle.properties file (which I'll do in my next commit since I've missed it for the first 3 dependencies here), it's not a problem to have duplicate dependencies. I think that we won't often remove dependencies and the version changes will be done in gradle.properties. What do you think?

Reply:
Sounds good.